staging/src/k8s.io/client-go/examples/dynamic-create-update-delete-deployment/main.go

```diff
@@ -0,0 +1,208 @@
+	//    2. Modify the "result" returned by Get() and retry Update(result) until
+	//       you no longer get a conflict error. This way, you can preserve changes made
+	//       by other clients between Create() and Update(). This is implemented below
+	//			 using the retry utility package included with client-go. (RECOMMENDED)
+	//
+	// More Info:
+	// https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#concurrency-control-and-consistency
+
+	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		// Retrieve the latest version of Deployment before attempting update
+		// RetryOnConflict uses exponential backoff to avoid exhausting the apiserver
+		result, getErr := client.Resource(deploymentRes).Namespace(namespace).Get("demo-deployment", metav1.GetOptions{})
+		if getErr != nil {
+			panic(fmt.Errorf("Failed to get latest version of Deployment: %v", getErr))
+		}
+
+		// update replicas to 1
+		if err := unstructured.SetNestedField(result.Object, int64(1), "spec", "replicas"); err != nil {
+			panic(fmt.Errorf("Failed to set replica value: %v", err))
+		}
+
+		// extract spec containers
+		containers, found, err := unstructured.NestedSlice(result.Object, "spec", "template", "spec", "containers")
+		if err != nil || !found || containers == nil {
+			panic(fmt.Errorf("Deployment containers not found or error in spec: %v", err))
+		}
+
+		// update container[0] image
+		if err := unstructured.SetNestedField(containers[0].(map[string]interface{}), "nginx:1.13", "image"); err != nil {
+			panic(err)
+		}
+		if err := unstructured.SetNestedField(result.Object, containers, "spec", "template", "spec", "containers"); err != nil {
+			panic(err)
+		}
+
+		_, updateErr := client.Resource(deploymentRes).Namespace(namespace).Update(result, metav1.UpdateOptions{})
+		return updateErr
+	})
+	if retryErr != nil {
+		panic(fmt.Errorf("Update failed: %v", retryErr))
+	}
+	fmt.Println("Updated deployment...")
+
+	// List Deployments
+	prompt()
+	fmt.Printf("Listing deployments in namespace %q:\n", apiv1.NamespaceDefault)
+	list, err := client.Resource(deploymentRes).Namespace(namespace).List(metav1.ListOptions{})
+	if err != nil {
+		panic(err)
+	}
+	for _, d := range list.Items {
+		replicas, found, err := unstructured.NestedInt64(d.Object, "spec", "replicas")
+		if err != nil || !found {
+			fmt.Printf("Replicas not found for deployment %s: error=%s", d.GetName(), err)
+			continue
+		}
+		fmt.Printf(" * %s (%d replicas)\n", d.GetName(), replicas)
+	}
+
+	// Delete Deployment
+	prompt()
+	fmt.Println("Deleting deployment...")
+	deletePolicy := metav1.DeletePropagationForeground
+	deleteOptions := &metav1.DeleteOptions{
+		PropagationPolicy: &deletePolicy,
+	}
+	if err := client.Resource(deploymentRes).Namespace(namespace).Delete("demo-deployment", deleteOptions); err != nil {
+		panic(err)
+	}
+
+	fmt.Println("Deleted deployment.")
+}
+
+func prompt() {
```

Comment:
nit: this function could be in a util package to be shared

Reply:
Again, the point example is to be an analogue of the typed example.